hack/tools/release/weekly/main.go

```diff
@@ -224,14 +203,82 @@ func trimTitle(title string) string {
 	return strings.TrimSpace(title)
 }
 
-type commit struct {
-	merge string
-	body  string
-}
-
 func formatMerge(line, prNumber string) string {
 	if prNumber == "" {
 		return line
 	}
-	return fmt.Sprintf("%s (%s)", line, prNumber)
+	return fmt.Sprintf("%s (#%s)", line, prNumber)
+}
+
+// Parse the time from string to time.Time format
+func parseTime(date string) (time.Time, error) {
+	datetime, err := time.Parse(timeLayout, date)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return datetime, nil
+}
+
+func getMergedPullRequests(client *github.Client, owner string, repo string, branch string, since time.Time, until time.Time) ([]*github.PullRequest, error) {
+	var allPullRequests []*github.PullRequest
+
+	// List up to 500 most recent pull requests to not overwhelm github
+	// This is a best effort alternative over paginating through all PRs
+	// as there are no filters for the time via sdk.
+	listOptions := &github.ListOptions{PerPage: 500}
```

Comment:
The [GitHub api docs](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests) mentions that the max page number is 100. I'm not sure why this doesn't cause an error but it might be better to use a slightly smaller pagesize.